plugins/completion/coq.nix

```
@@ -5,56 +5,74 @@
   pkgs,
   ...
 }:
-with lib; let
-  cfg = config.plugins.coq-nvim;
-in {
-  options = {
-    plugins.coq-nvim = {
-      enable = mkEnableOption "coq-nvim";
+with lib;
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "coq-nvim";
+    originalName = "coq_nvim";
+    defaultPackage = pkgs.vimPlugins.coq_nvim;
 
-      package = helpers.mkPackageOption "coq-nvim" pkgs.vimPlugins.coq_nvim;
+    maintainers = [maintainers.traxys];
 
+    extraOptions = {
       installArtifacts = mkEnableOption "and install coq-artifacts";
-
-      autoStart = mkOption {
-        type = with types; nullOr (oneOf [bool (enum ["shut-up"])]);
-        default = null;
-        description = "Auto-start or shut up";
+      artifactsPackage = mkOption {
+        type = types.package;
+        description = "Package to use for coq-artifacts (when enabled with installArtifacts)";
+        default = pkgs.vimPlugins.coq-artifacts;
       };
+    };
 
-      recommendedKeymaps = mkOption {
-        type = with types; nullOr bool;
-        default = null;
-        description = "Use the recommended keymaps";
-      };
+    # Introduced 12-03-2022, remove 12-05-2022
```

```suggestion
    # TODO: Introduced 12-03-2022, remove 12-05-2022
```

Otherwise we might forget to remove those...